File: app/src/main/java/com/example/testapp/MainActivity.java
package com.example.testapp;
import androidx.activity.ComponentActivity;
import android.os.Bundle;
import android.os.SystemClock;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends ComponentActivity {

    private CheckBox checkBoxHarry, checkBoxMatrix, checkBoxJoker;
    private RadioGroup rgMaritalStatus;
    private ProgressBar progressBar;

    @Override
    protected void onCreate (Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        checkBoxHarry = findViewById(R.id.checkboxHarry);
        checkBoxMatrix = findViewById(R.id.checkboxMatrix);
        checkBoxJoker = findViewById(R.id.checkboxJoker);

        rgMaritalStatus = findViewById(R.id.rgMaritalStatus);

        progressBar = findViewById(R.id.progressBar);

        Thread thread = new Thread(new Runnable() {
            @Override
            public void run() {
                for(int i=0; i<10; i++){
                    progressBar.incrementProgressBy(10);
                    SystemClock.sleep(500);
                }
            }
        });
        thread.start();

        int checkedButton = rgMaritalStatus.getCheckedRadioButtonId();
        if(checkedButton == R.id.rbMarried){
            Toast.makeText(MainActivity.this, "Married", Toast.LENGTH_SHORT).show();
        }else if (checkedButton == R.id.rbSingle){
            Toast.makeText(MainActivity.this, "Single", Toast.LENGTH_SHORT).show();
        }else if (checkedButton == R.id.inRel){
            Toast.makeText(MainActivity.this, "In a Relationship", Toast.LENGTH_SHORT).show();

        }
        rgMaritalStatus.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                if(checkedId == R.id.rbMarried){
                    Toast.makeText(MainActivity.this, "Married", Toast.LENGTH_SHORT).show();
                }else if (checkedId == R.id.rbSingle){
                    Toast.makeText(MainActivity.this, "Single", Toast.LENGTH_SHORT).show();
                }else if (checkedId == R.id.inRel){
                    Toast.makeText(MainActivity.this, "In a Relationship", Toast.LENGTH_SHORT).show();
                }
            }
        });

        if(checkBoxHarry.isChecked()){
            Toast.makeText(MainActivity.this, "You have watched harry potter", Toast.LENGTH_SHORT).show();
        }else{
            Toast.makeText(MainActivity.this, "WATCH the potter", Toast.LENGTH_SHORT).show();
        }

        checkBoxHarry.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if(isChecked){
                    Toast.makeText(MainActivity.this, "You have watched harry potter", Toast.LENGTH_SHORT).show();
                }else{
                    Toast.makeText(MainActivity.this, "WATCH the potter", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: app/src/main/java/com/example/testapp/MainActivity.kt
package com.example.testapp

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import android.widget.TextView
import android.widget.EditText
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.testapp.ui.theme.TestAppTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }


    @SuppressLint("SetTextI18n")
    fun onBtnClick(view: View) {
        val first: EditText = findViewById(R.id.firstName)
        val last: EditText = findViewById(R.id.lastName)
        val email: EditText = findViewById(R.id.emailText)
        val firstOutput: TextView = findViewById(R.id.fOutput)
        val lastOutput: TextView = findViewById(R.id.lOutput)
        val emailOutput: TextView = findViewById(R.id.eOutput)
        firstOutput.text = "First Name: ${first.text.toString()}"
        lastOutput.text = "Last Name: ${last.text.toString()}"
        emailOutput.text = "Email: ${email.text.toString()}"
    }
}